Slim/Control/Queries.pm

```diff
@@ -857,21 +861,45 @@ sub albumsQuery {
 				$request->addResultLoopIfValueDefined($loopname, $chunkCount, 'textkey', $textKey);
 			}
 
+			# Override $contributorSql if we're dealing with a Work: output Artist, Orchestra, Conductor in that order.
+			if ( defined $work ) {
+				my @roles = ( 'ARTIST', 'BAND', 'CONDUCTOR' );
+				$contributorSql = sprintf( qq{
+					WITH temp as (
+					SELECT CASE WHEN contributor_track.role = 1 THEN 'ARTIST' WHEN contributor_track.role = 3 THEN 'CONDUCTOR' WHEN contributor_track.role = 4 THEN 'BAND' END role,
```

Comment:
Working on my laptop's screen I lack the space to review this right now... can you elaborate what this mapping is for? This IMHO is complicating the query unnecessarily, as the mapping will/can be done later on easily.

Could you please format the SQL to make it more readable?

Reply:
I need something to "ORDER BY" that gives the correct order - artists, band(orchestra), conductor. By chance the actual role names do that (ABC!).

The SQL is working in two stages:

1. GROUP_CONCAT distinct artist ids/names for each role - order by as above.
2. this feeds in to the second stage, which uses GROUP_CONCAT to output a single row of Artist,Band,Conductor in that order.

Later versions of sqlite allow an ORDER BY within a GROUP_CONCAT (and other aggregate functions), but not the one we're using.

I've reformatted the SQL as requested.

Reply:
Thanks for the additional information (and the formatting, which makes this so much easier to read)!

But do we really need that `CASE` statement, as the numerical role values would result in the same sort order anyway? Seems to only complicate the query for no reason?